packages/gatsby/src/utils/webpack-utils.ts

```diff
@@ -416,25 +416,49 @@ export const createWebpackUtils = (
     } = {}): RuleSetRule => {
       return {
         test: /\.(js|mjs|jsx)$/,
-        include: (modulePath: string): boolean => {
-          // when it's not coming from node_modules we treat it as a source file.
-          if (!vendorRegex.test(modulePath)) {
-            return true
-          }
-
-          // If the module uses Gatsby as a dependency
-          // we want to treat it as src so we can extract queries
-          return modulesThatUseGatsby.some(module =>
-            modulePath.includes(module.path)
-          )
-        },
-        type: `javascript/auto`,
-        use: [
-          loaders.js({
-            ...options,
-            configFile: true,
-            compact: PRODUCTION,
-          }),
+        oneOf: [
+          /**
+           * Handle page templates separately to scope certain babel transforms appropriately
+           * (e.g. removing special module exports named `config` and `getServerData` in
+           * `babel-plugin-remove-api`).
+           */
+          {
+            issuer: /async-requires/, // Only JS imports in async-requires are for page templates
```

Comment:
I'm curious if you figured out why query didn't work.

Reply:
It turns out that most of the same tests fail for both approaches, with the exception of `resourceQuery` failing one more (`e2e_tests_visual-regression`).

- `resourceQuery` test run - https://app.circleci.com/pipelines/github/gatsbyjs/gatsby/78921/workflows/ee57ade6-349a-4ac8-b187-d2922e5d785e
- `issuer` test run - https://app.circleci.com/pipelines/github/gatsbyjs/gatsby/78922/workflows/5b101937-d160-45b9-9b68-1c9aaea43595

So query works per se. I didn't check why that visual test breaks since issuer should effectively achieve the same thing without hard coding in a query.

Reply:
Can we make it a making moves topic, it's good to understand why this is different behavior cause it doesn't make sense.

Reply:
Yes.